modules/apps/portal-workflow/portal-workflow-test/src/testFunctional/macros/Workflow.macro

```diff
@@ -979,6 +979,60 @@ definition {
 			value1 = "Within");
 	}
 
+	macro updateDueDateAssignedToMyRolesFutureDate {
+		var key_workflowAssetTitle = "${key_workflowAssetTitle}";
+		var key_workflowTask = "${key_workflowTask}";
+		var displayDateDay = DateUtil.getCurrentDate();
+		var displayDateMonth = DateUtil.getCurrentMonth();
+		var currentYear = DateUtil.getCurrentYear();
+		var futureDay = ${displayDateDay} + 1;
+		var futureMonth = ${displayDateMonth} + 1;
+		var futureYear = ${currentYear} + 1;
+		timePeriod = "${key_timePeriod}";
+
+		Workflow.viewTaskAssignedToMyRoles(
+			ddlDataDefinitionName = "${ddlDataDefinitionName}",
+			workflowAssetTitle = "${key_workflowAssetTitle}",
+			workflowAssetType = "${workflowAssetType}",
+			workflowTask = "${key_workflowTask}");
+
+		Workflow.gotoTaskAssignedToMyRolesActionsCmd(
+			workflowAssetTitle = "${key_workflowAssetTitle}",
+			workflowTask = "${key_workflowTask}");
+
+		MenuItem.click(menuItem = "Update Due Date");
+
+		SelectFrame(locator1 = "MyWorkflowTasks#IFRAME_UPDATE_DUE_DATE");
+
+		SelectFrame(locator1 = "WorkflowDefinition#DIALOG_BOX_BODY");
+
+		SelectFrame(locator1 = "IFrame#IFRAME");
+
+		if ("${key_timePeriod}" == "day") {
+			Type(
+				locator1 = "MyWorkflowTasks#DIALOG_BOX_DUE_DATE",
+				value1 = "${displayDateMonth}/${futureDay}/${displayDateYear}");
+
+			Type(
+				locator1 = "MyWorkflowTasks#DIALOG_BOX_DUE_DATE_TIME",
+				value1 = "07:30 PM");
+		}
+
+		if ("${key_timePeriod}" == "month") {
+			Type(
+				locator1 = "MyWorkflowTasks#DIALOG_BOX_DUE_DATE",
+				value1 = "${futureMonth}/${displayDateDay}/${displayDateYear}");
+		}
+
+		if ("${key_timePeriod}" == "year") {
+			Type(
+				locator1 = "MyWorkflowTasks#DIALOG_BOX_DUE_DATE",
+				value1 = "${displayDateMonth}/${displayDateDay}/${futureYear}");
+		}
+
+		Workflow.confirmWorkflowAction();
+	}
```

Comment:
Maybe you can use the macro you already have created at this [PR](https://github.com/rodrigocunhaa/liferay-portal/pull/835/commits/d299c9d40897bf65014e97b95381b3a57e33cbec). It's more general and i believe it will work. Thank you.